### ISSUE TITLE
Train, test and validation in modelling

### ISSUE THREAD

**Issue by JGarciaCondado** [compneurobilbao/ageml#31]
The software package already has minimum cross validation strategies implemented but there are a few adjustments to make.

- There should be the option to split the data into train, validation and test. Currently controls are all used in a CV strategy and there is no separate test group.
- Another option is to allow users to choose the CV strategy by using the import command (e.g. “-cv from sklearn.cross_validation import LeaveOneOut”)
- The training, validation and test ids of each subject should be saved as a .txt for reproducibility.
- In classification currently there is no warning if the dataset is unbalanced. A warning should be thrown and if possible the user should be given the option to balance the dataset by simple subsampling the largest group to reduce it to the smallest group. A heuristic approach for unbalanced could be 25% more data in one group than the other or something along those lines.

**Comment by JGarciaCondado** [comment]
We also have a problem with how we do the cross validation. This should be done like the following for each of the cases.

1. No hypeparameter tunning:

We have cross validation and get train and validation metrics as well as deltas for each of the subjects in the validation set. Then we apply to test and show test.

2. Hyperparameter tunning:

Use grid search on validation set to obtain best hyperparameters and from this obtain both the train and val scores as well as the predictions. Then apply best model to test.

3. Hyperopt:

Split into train and test. Apply hyperopt to train and then apply to test.

At the end in all train the model with all CN for further predictions with others. 

We should somehow notify which controls are from validation and which are for test.

**Comment by JGarciaCondado** [comment]
We have move the need to warn about unbalanced datasets to a different issue. Also implementing different CV strategies can become quite complicated and does not add much extra value.
